### DISCUSSION TITLE
Hints for adding ApplicationComponents

### DISCUSSION THREAD

**Discussion by yotommy** [playframework/playframework#12335] [Q&A] [answered]
I work on [CiviForm](http://civiform.us/), an open-source project that simplifies the application process for government benefits programs by re-using applicant data for multiple benefits applications.

I am investigating the use of [play-secret-rotation](https://github.com/guardian/play-secret-rotation) in our (Java) server. The [`README`](https://github.com/guardian/play-secret-rotation/blob/main/aws-parameterstore/README.md) contains these instructions:

> In your `ApplicationComponents`, mix-in `RotatingSecretComponents` and provide the `secretStateSupplier` required by that trait.

As mentioned, our server uses Java (and Play 2.8). Can anyone point me to some resource that helps me understand how/where to do this in Java? As far as I can tell, our project does not contain `ApplicationComponents` at present. Our server code is here: https://github.com/civiform/civiform

Thanks in advance.

**Comment by ihostage** [comment, marked as answer]
Hi, @yotommy! `ApplicationComponents` is a part of [compile-time dependency injection](https://www.playframework.com/documentation/2.8.x/JavaCompileTimeDependencyInjection#Compile-Time-Dependency-Injection) that also supported by Play. If you use Guice as a runtime DI, you won't need a components.

**Reply by yotommy** [reply]
Great, that clears up a lot! Now I just have to figure out what to provide/inject. =;^)

Thanks,
Tom
